### PR TITLE
Pascal's triangle solution - one with for loops and one with ranges

### PR DESCRIPTION
**Pull request by makcgr**
First I solved it with for loops.

Then I looked up the "pro" solution in Python and it used ranges.
I did it with ranges too, but in C#. It was a bit slower than the first one.

### CODE REVIEW

File: mgrachev/pascals triangle/Program.cs [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+}
+
+public class Solution 
+{
+    // Solution: 
+    // We build "Pascal's triangle" jagged array (array of arrays),
+    // starting from the top of triangle.
+    // Each next inner array is [n+1] elements when compared to the previous array of [n]
+    // From 3rd row on we begin to form values of current row as the sum of above's "left" and "right" values.
+    // Time complexity is O(N^2) since we iterate two times and space complexity is also O(N^2)
```
Comment (PShchegolevatykh) [inline mgrachev/pascals triangle/Program.cs:36]: Thanks for mentioning complexity explicitly here!

File: mgrachev/pascals triangle/Program.cs [hunk 2]
```diff
@@ -0,0 +1,89 @@
...
+        int[][] triangleArray = new int[numRows][];
+
+        for (int num = 1; num <= numRows; num++)
+        {
+            triangleArray[num-1] = new int[num];    // we are row with index [num-1]
+            var currentRow = triangleArray[num-1];
+            
+            for(int x=0; x<=num-1; x++)
+            {                
+                if(num==1 || num==2)
```
Comment (PShchegolevatykh) [inline mgrachev/pascals triangle/Program.cs:49]: Takes a while to understand why num==2 gives us 1 as a result, but maybe it's just me. :)

File: mgrachev/pascals triangle/Program.cs [hunk 3]
```diff
@@ -0,0 +1,89 @@
...
+                    var right = x > aboveRow.Length-1 ? 0 : aboveRow[x];  // number above and to the right
+
+                    currentRow[x] = left + right;
+                }                    
+            }
+        }
+        return triangleArray;
+    }
+
+    public IList<IList<int>> Generate2(int numRows)
```
Comment (PShchegolevatykh) [inline mgrachev/pascals triangle/Program.cs:67]: Mind-blowing, thanks for this variant!

**Review by PShchegolevatykh — APPROVED** [review]

File: mgrachev/pascals triangle/Program.cs (hunk 2 above)
Comment (makcgr) [reply mgrachev/pascals triangle/Program.cs:49]: Yep, probably should've left a comment here:

Numbers 1 and 2, it's the top two rows of triangle:

```
num=1              1
num=2          1      1
...
```

**Pull request merged**